cpp/src/search/search.cu

```diff
@@ -210,6 +212,62 @@ bool contains(column_view const& col,
                                              col, value,
                                              stream, mr);
 }
+
+struct multi_contains_dispatch {
+  template <typename Element>
+  void operator()(column_view const& col,
+                  column_view const& in_col,
+                  mutable_column_view &result_view,
+                  cudaStream_t stream) {
+
+    auto hash_set = fixed_hash_set<Element>::create(in_col, stream);
+    auto device_hash_set = hash_set.to_device();
+
+    auto d_col_ptr = column_device_view::create(col, stream);
+    auto d_col = *d_col_ptr;
+
+    thrust::transform(rmm::exec_policy(stream)->on(stream),
+                      thrust::make_counting_iterator<size_type>(0),
+                      thrust::make_counting_iterator<size_type>(col.size()),
+                      result_view.begin<bool8>(),
+                      [device_hash_set, d_col] __device__ (size_t index) {
+                        return d_col.is_null(index) || device_hash_set.contains(d_col.element<Element>(index));
+                      });
+  }
+};
+
+std::unique_ptr<column> contains(column_view const& col,
+                                 column_view const& in_col,
+                                 rmm::mr::device_memory_resource* mr,
+                                 cudaStream_t stream = 0) {
+
+  CUDF_EXPECTS(col.type() == in_col.type(), "DTYPE mismatch");
+
+  std::unique_ptr<column> result = make_numeric_column(data_type{experimental::type_to_id<bool8>()},
+                                                       col.size(),
+                                                       copy_bitmask(col),
+                                                       col.null_count(),
+                                                       stream, mr);
+  
+  if (col.size() == 0) {
+    return result;
+  }
+
+  mutable_column_view result_view = result.get()->mutable_view();
+
+  if (in_col.size() == 0) {
+    bool8 f = false;
+    thrust::fill(rmm::exec_policy(stream)->on(stream), result_view.begin<bool8>(), result_view.end<bool8>(), f);
+    return result;
+  }
+
+  cudf::experimental::type_dispatcher(col.type(),
```

Comment:
I think your dispatcher functor should return the output column rather than taking a mutable column view...  @jrhemstad gave this advice (on slack) the other day: 

PSA: I'm still seeing a common pattern that should be avoided when possible.
```
unique_ptr<column> output(...);
mutable_column_view output_view(output);

type_dispatcher(type, functor{}, output_view, ...);
```

I.e., outputs are being pre-allocated and then passed as mutable views. This pattern precludes adding string support. Instead, do this:

```
unique_ptr<column> output;

output = type_dispatcher(type, functor{}, ...);
```

The output column is returned. This allows adding a string_view specialization to the functor that returns the appropriate output column.

Reply:
I considered that.  I'm fine with changing it if you would prefer, it would make things a bit more consistent with the other code we're developing.  In this particular case, the output is always a fixed_width column of bool8.  The issue with this idiom is if the output column has to be strings, since you can't populate a string column in this way.  So in this case there can be no problem with using this idiom.

But if we think the consistency is more important I can make that change.

Reply:
In general I still prefer to see outputs returned, but it's not as big of deal in this case since the output is always `bool8`.